packages/zosjobs/src/SearchJobs.ts

```diff
@@ -0,0 +1,162 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, ImperativeError, IRestClientResponse, IOptionsFullResponse, Headers, Logger} from "@zowe/imperative";
+import { ZosmfRestClient, ZosmfHeaders } from "@zowe/core-for-zowe-sdk";
+import { GetJobs} from "./GetJobs";
+import { ISearchJobsParms } from "./doc/input/ISearchJobsParms";
+import { IJobFile } from "./doc/response/IJobFile";
+import { IJob } from "./doc/response/IJob";
+import { JobsConstants } from "./JobsConstants";
+import { ZosJobsMessages } from "./JobsMessages";
+
+/**
+ * Class to handle the searching of z/OS batch job spool output
+ * @export
+ * @class SearchJobs
+ */
+export class SearchJobs {
+    /**
+     * Search Jobs - Search the spool output of the specified job name for a string or regular expression
+     * @static
+     * @param {AbstractSession} session - z/OSMF connection info
+     * @param {ISearchJobsParms} searchParms - The search parameters for the API call
+     * @returns {Promise<string>} - promise that resolves to spool output
+     * @throws {ImperativeError} --search-string or --search-regx must be specified
+     * @memberof searchJobs
+     */
+    public static async searchJobs(session: AbstractSession, searchParms : ISearchJobsParms) {
+        const jobName: string = searchParms.jobName;
+        const searchString: string= searchParms.searchString;
+        const searchRegex: string = searchParms.searchRegex;
+        const caseInsensitive: boolean = searchParms.caseInsensitive;
+        const searchLimit: number = searchParms.searchLimit;
+        const fileLimit: number = searchParms.fileLimit;
+
+        Logger.getAppLogger().info("SearchJobs.searchJobs() called!");
+        let replyBuffer:string = "";
+
+        // Validate that a search string or regex parameter was passed
+        if(searchRegex === undefined && searchString === undefined){
+            throw new ImperativeError({ msg: ZosJobsMessages.missingSearchOption.message });
```

Comment:
Curious if we should use ImperativeExpect here (and in some of the others below)

I'm not sure if the syntax will be any better since we are talking about two parameters that should be both undefined.
That's why I'm not requesting changes with this comment 😋

Reply:
I hope the way I worded the message makes since. I used the same message as it states that you need to specify either one or the other. If neither is specified or both are specified then it is a failing condition.

Reply:
yeah, I do think it makes sense 😋